src/topology/mod.rs

```diff
@@ -361,12 +370,11 @@ impl RunningTopology {
         let task = handle_errors(task.instrument(span.clone()), self.abort_tx.clone());
         let spawned = oneshot::spawn(task, &rt.executor());
         if let Some(previous) = self.tasks.insert(name.to_string(), spawned) {
-            previous.forget();
+            previous.forget(); // todo should we really be forgetting here?
```

Comment:
Good question. We should try to figure out why we were doing this in the first place.

Reply:
Was this done so that we can spawn the new source before fully shutting down the previous one? For example, if we did the takeover source below would dropping the original source future cause issues there? It would be good to figure this out.

Reply:
I filed https://github.com/timberio/vector/issues/2092 for this issue and will remove this TODO for now.